agent/node.go

```diff
@@ -240,28 +240,30 @@ func (n *Node) run(ctx context.Context) (err error) {
 		return err
 	}
 
-	if n.role == ca.ManagerRole {
-		n.managerRoleCh <- struct{}{}
-	}
-
 	forceCertRenewal := make(chan struct{})
 	go func() {
-		n.RLock()
-		lastRole := n.role
-		n.RUnlock()
 		for {
 			select {
 			case <-ctx.Done():
 				return
 			case apirole := <-n.roleChangeReq:
```

Comment:
I think we should only send on this channel if there was a role change in the first place. Why do we have the "has this role changed" logic inside of this case, and not futher up the stack?